core-bundle/contao/drivers/DC_Table.php

```diff
@@ -3205,109 +3205,109 @@ protected function save($varValue)
+			$blnVersionize = false;
 
-			// If the field is a fallback field, empty all other columns (see #6498)
-			if ($varValue && ($arrData['eval']['fallback'] ?? null))
+			foreach ($arrValues as $strField => $varValue)
 			{
-				$varEmpty = Widget::getEmptyValueByFieldType($GLOBALS['TL_DCA'][$this->strTable]['fields'][$this->strField]['sql'] ?? array());
-				$arrType = array_filter(array($GLOBALS['TL_DCA'][$this->strTable]['fields'][$this->strField]['sql']['type'] ?? null));
+				$arrData = $GLOBALS['TL_DCA'][$this->strTable]['fields'][$strField] ?? array();
 
-				if (($GLOBALS['TL_DCA'][$this->strTable]['list']['sorting']['mode'] ?? null) == self::MODE_PARENT)
-				{
-					$this->Database
-						->prepare("UPDATE " . $this->strTable . " SET " . Database::quoteIdentifier($this->strField) . "=? WHERE pid=?")
-						->query('', array($varEmpty, $currentRecord['pid'] ?? null), $arrType);
-				}
-				else
+				// If the field is a fallback field, empty all other columns (see #6498)
+				if ($varValue && ($arrData['eval']['fallback'] ?? null))
 				{
-					$this->Database
-						->prepare("UPDATE " . $this->strTable . " SET " . Database::quoteIdentifier($this->strField) . "=?")
-						->query('', array($varEmpty), $arrType);
-				}
-			}
+					$varEmpty = Widget::getEmptyValueByFieldType($GLOBALS['TL_DCA'][$this->strTable]['fields'][$strField]['sql'] ?? array());
+					$arrType = array_filter(array($GLOBALS['TL_DCA'][$this->strTable]['fields'][$strField]['sql']['type'] ?? null));
 
-			$arrTypes[] = $GLOBALS['TL_DCA'][$this->strTable]['fields'][$this->strField]['sql']['type'] ?? null;
-
-			if (!isset($arrData['eval']['versionize']) || $arrData['eval']['versionize'] !== false)
-			{
-				$blnVersionize = true;
-			}
+					if (($GLOBALS['TL_DCA'][$this->strTable]['list']['sorting']['mode'] ?? null) == self::MODE_PARENT)
+					{
+						$this->Database
+							->prepare("UPDATE " . $this->strTable . " SET " . Database::quoteIdentifier($strField) . "=? WHERE pid=?")
+							->query('', array($varEmpty, $currentRecord['pid'] ?? null), $arrType);
+					}
+					else
+					{
+						$this->Database
+							->prepare("UPDATE " . $this->strTable . " SET " . Database::quoteIdentifier($strField) . "=?")
+							->query('', array($varEmpty), $arrType);
+					}
+				}
 
-			$this->varValue = StringUtil::deserialize($varValue);
+				$arrTypes[] = $GLOBALS['TL_DCA'][$this->strTable]['fields'][$strField]['sql']['type'] ?? null;
 
-			if (\is_object($this->objActiveRecord))
-			{
-				$this->objActiveRecord->{$this->strField} = $this->varValue;
-			}
-		}
+				if (!isset($arrData['eval']['versionize']) || $arrData['eval']['versionize'] !== false)
+				{
+					$blnVersionize = true;
+				}
 
-		if (!empty($arrValues))
-		{
-			$arrValues['tstamp'] = time();
+				// Update the active record and current field/value (backwards compatibility)
+				$varValue = StringUtil::deserialize($varValue);
+				$this->varValue = $varValue;
+				$this->strField = $strField;
```

Comment:
Why setting `$this->varValue` and `$this->strField` here inside the foreach? There is no external code called inside the loop so these should not be necessary.
I think only the `$this->objActiveRecord->{$strField} = StringUtil::deserialize($varValue);` below is needed for BC.

Reply:
Would you rather set them after the loop like so or completely remove the lines?
```php
$this->strField = array_key_last($arrValues);
$this->varValue = $arrValues[$this->strField];
```

Reply:
You mean like if someone relies on the fact that these are set to the last item when executing the `onsubmit_callback`?

But even then, this would already have occured on line 2007, so I’d completely remove them here.

Reply:
Agree. Adjusted in 13c5f0f2cd3ddc4e1a408074a2141a05651d7170.